blog/2025-07-01-safe-deployment-of-versioned-workflows/2025-07-01-safe-deployment-of-versioned-workflows.md

```diff
@@ -0,0 +1,257 @@
+
+### Deployment with Dynamic Configuration
+
+Using the new options adds an extra step, which can lead to increased deployment time. If your service has a dynamic configuration, you can integrate it with the new functions to eliminate this problem and maintain the same number of deployments.
+
+The solution combines the code changes from [Step 1](https://docs.google.com/document/d/1mNR3BPqX94dY4-Hd_vBGe2es4XpIbylrE0TCH8xl7Fc/edit?tab=t.0#bookmark=id.rnkz0efm7hc1) and [Step 2](https://docs.google.com/document/d/1mNR3BPqX94dY4-Hd_vBGe2es4XpIbylrE0TCH8xl7Fc/edit?tab=t.0#bookmark=id.txibbvf2qjv) into a single code change. [Step 2](https://docs.google.com/document/d/1mNR3BPqX94dY4-Hd_vBGe2es4XpIbylrE0TCH8xl7Fc/edit?tab=t.0#bookmark=id.txibbvf2qjv) becomes a change in Dynamic Configuration rather than a new version deployment. Therefore, in this case, the deployment will include the following steps:
+
+#### Step 1
+
+This Step 1 is similar to the original [Step 1](https://docs.google.com/document/d/1mNR3BPqX94dY4-Hd_vBGe2es4XpIbylrE0TCH8xl7Fc/edit?tab=t.0#bookmark=id.rnkz0efm7hc1) but introduces the retrieval of a value from the Dynamic Configuration. To achieve forward and backward compatibility of the change, the Dynamic Configuration must have the value of the minimum support version \- `workflow.DefaultVersion` at this step.
+
+```go
+
+var (
+	// Get dynamic config client during app initialization 
+dcClient, _ 	= dynamicConfig.NewClient()
+region 	= getRegion()
+)
+
+// Git tag: v0.2   
+// MyWorkflow supports: workflow.DefaultVersion and 1
+func MyWorkflow(ctx workflow.Context) error {   
+// The call of the dynamic configuration client is safe for the workflow execution
+	// because it's not saved in the history events. 
+    changeIDExecVersion, err := dcClient.GetIntValue("changeId", map[string]interface{
+	"region": region,
+})
+if err != nil {
+	// Error handling must be non-deterministic 
+}
+    
+    // When GetVersion is executed for the first time, changeIDExecVersion will be returned
+    version := workflow.GetVersion(ctx, "MyChange", workflow.DefaultVersion, 1, workflow.ExecuteWithVersion(changeIDExecVersion))
+    
+    if version == workflow.DefaultVersion {
+       return workflow.ExecuteActivity(ctx, FooActivity).Get(ctx, nil)
+    } 
+    
+    return workflow.ExecuteActivity(ctx, BarActivity).Get(ctx, nil)
+  }
+
+
+```
+
+#### Step 2
+
+Once all v0.2 workers are replaced by v0.1 workers, you need to change the Dynamic Configuration value to the next version (in this case, 1). This will activate the new logic for new workflow executions. To roll back from Step 2 to Step 1, simply revert the dynamic configuration value.
```

Comment:
```suggestion
Once all v0.2 workers are replaced by v0.1 workers, you will need to change the Dynamic Configuration value to the next version (in this case, 1). This will activate the new logic for new workflow executions. To roll back from Step 2 to Step 1, simply revert the dynamic configuration value.
```